### PR TITLE
Restore Clippy cleanliness

### PR DESCRIPTION
**Pull request by mkeeter**
The compiler bump brought along a bunch of new Clippy lints!

I've gone through and fixed everything for Gimlet / Sidecar / PSC.

### CODE REVIEW

File: task/thermal/src/control.rs [hunk 1]
```diff
@@ -339,17 +339,11 @@ impl OneSidedPidState {
         } else {
             (-out_pd, output_limit - out_pd)
         };
-        self.integral = self.integral.max(integral_min).min(integral_max);
+        self.integral = self.integral.clamp(integral_min, integral_max);
```
Comment (jgallagher) [inline task/thermal/src/control.rs:342]: TIL about `.clamp()`!

**Review by andrewjstone — APPROVED** [review]
Thanks for fixing this and cleaning up my mess Matt!

File: drv/sprot-api/src/lib.rs [hunk 2]
```diff
@@ -481,23 +489,27 @@ pub struct RxMsg {
...
+
 impl RxMsg {
     pub fn new() -> RxMsg {
         RxMsg { buf: [0; BUF_SIZE] }
     }
 
-    pub fn len(&self) -> usize {
-        self.buf.len()
-    }
-
```
Comment (jgallagher) [inline drv/sprot-api/src/lib.rs:492]: I don't think `len()` is provided by `AsMut<_>` is it?

Comment (jgallagher) [reply drv/sprot-api/src/lib.rs:492]: Nevermind, you fixed this already

Comment (mkeeter) [reply drv/sprot-api/src/lib.rs:492]: No, I'm removing this entirely to fix a warning about the lack of `is_empty`.

(you can still get the length with `as_slice().len()`, which is what we do in `TxBuf` already)

**Pull request merged**